tests/testthat/test-namespace.R

```diff
@@ -418,6 +427,23 @@ test_that("Invalid imports throw a helpful error", {
   expect_snapshot(out <- roc_proc_text(namespace_roclet(), block))
   expect_equal(out, "importFrom(AnUnknownUnavailablePackage,Unchecked)")
 
+  # make sure to match several forms of non-syntactic names
+  expect_no_warning(expect_equal(
+    roc_proc_text(namespace_roclet(), "#' @importFrom stringr %>%\nNULL"),
+    'importFrom(stringr,"%>%")'
+  ))
```

Comment:
This is a testing philosophy question — should you test `namspace_roclet()` for each of the quoting options or is it better to test one type of quotes for the whole flow and then rely on testing  `normalize_quotes()`/`strip_quotes()` for the others? I'd tend to think the later, but I'd love to know what you think.

Reply:
I lean to be as verbose/comprehensive as possible, even if it's very obvious from looking at the _current_ code that we're covered. My thinking for adding all of them is it guards against any eventual refactoring that forgets to cover all four options for quoting.

One thing we use to reduce to repetitiveness is condense things into a [{patrick}](https://cran.r-project.org/web/packages/patrick/index.html) test.

Reply:
Patrick looks interesting, although I immediately want to tweak the syntax from:

```R
with_parameters_test_that("Data is successfully converted:", {
    input <- convert(test_data)
    expect_type(input, type)
  },
  test_data = list(numeric_data, character_data),
  type = c("double", "character"),
  .test_name = type
)
```

to:

```R
test_these("{type} data is successfully converted:", 
  {
    input <- convert(test_data)
    expect_type(input, type)
  },
  list(
    double = list(test_data = numeric_data, type = "double"),
    character = list(test_data = character_data, type = "character")
  )
)
```

I do wonder if testthat should incorporate patrick-like functionality since it is a pretty common need.

Reply:
cc @michaelquinn32. I do see some old threads in {testthat}: #962, #563. Should we revive one of those, or start a new issue?

`test_these()` does sound pretty nice as a function name. Glad you found https://github.com/google/patrick/issues/18 already, I was going to link that.

Reply:
Let's keep talking in patrick, and see where we get.